buildSrc/src/test/kotlin/com/example/model/Version.kt

```diff
@@ -30,12 +35,51 @@ class Version {
         fun fromJson(serializedObject: String): Version {
             try {
                 val jsonObject = JsonParser.parseString(serializedObject).asJsonObject
-                return Version()
+                val id = jsonObject.get("id").toString().let {
+                    Id.fromJson()
+                }
+                val date = jsonObject.get("date")?.toString()?.let {
+                    Date.fromJson()
+                }
+                return Version(id, date)
             } catch (e: IllegalStateException) {
                 throw JsonParseException(e.message)
             } catch (e: NumberFormatException) {
                 throw JsonParseException(e.message)
             }
         }
     }
+
+    class Id {
+        val serialNumber: Double = 12112.0
+
+        fun toJson(): JsonElement {
+            val json = JsonObject()
+            json.addProperty("serialNumber", serialNumber)
+            return json
+        }
+
+        companion object {
+            @JvmStatic
+            fun fromJson(): Id = Id()
```

Comment:
It is slightly useless (and misleading) to have a method that basically just call the the constructor. We could simply just not create this method and call the constructor directly, WDYT?

Reply:
Like above, I would like to keep it consistent, otherwise for a class which has constant primitive properties + a constant object we will have this method and in this case for a class which only has constant primitives we will not have it. This will create more noise in the deserializer code generator.

Reply:
Actually this is ugly :(. Let me see what I can do to remove that.